src/lib/spinel/radio_spinel_impl.hpp

```diff
@@ -2232,6 +2232,8 @@ void RadioSpinel<InterfaceType, ProcessContextType>::HandleRcpTimeout(void)
 {
 #if OPENTHREAD_SPINEL_CONFIG_RCP_RESTORATION_MAX_COUNT > 0
     mRcpFailed = true;
+#elif OPENTHREAD_SPINEL_CONFIG_RCP_CUSTOM_RESTORATION
+    mSpinelInterface.HandleRcpFailed();
```

Comment:
I wonder if this strategy has to co-work with RCP restoration?
Otherwise, how do we restore RCP states as in `RecoverFromRcpFailure`? 

Thoughts? @gjc13 @jinran-google

Reply:
Looks like they should be mutually exclusive. I think letting the custom handler do all the work without any additional handling should provide the most flexibility.

Reply:
How can the custom handler restore RCP states if the RCP restoration is not executed? 

I think the purpose of custom handler is more about resetting the hardware device. We would still need RCP restoration to restore the software states? @jinran-google @gjc13

Reply:
Added `SuccessOrExit` to `HandleRcpFailed`. If the custom handler succeeds we'll proceed with the restoration.